gqt_test.go

```diff
@@ -7,6 +7,363 @@ import (
+			},
+			name:  "ConstraintLenLess",
+			value: uint(47),
+		},
+		{
+			input: gqt.ConstraintLenGreaterOrEqual{
+				Value: 404,
+			},
+			name:  "ConstraintLenGreaterOrEqual",
+			value: uint(404),
+		},
+		{
+			input: gqt.ConstraintLenLessOrEqual{
+				Value: 404,
+			},
+			name:  "ConstraintLenLessOrEqual",
+			value: uint(404),
+		},
+	} {
+		t.Run(td.name, func(t *testing.T) {
+			name, value, err := gqt.ConstraintNameAndValue(td.input)
+			require.Nil(t, err)
+			require.Equal(t, td.name, name)
+			require.Equal(t, td.value, value)
+		})
+	}
+}
+
+func TestConstraintName(t *testing.T) {
+	for _, td := range []struct {
+		input  BasicConstraintInterface
+		expect string
+	}{
+		{
+			input:  gqt.ConstraintOr{},
+			expect: "ConstraintOr",
+		},
+		{
+			input:  gqt.ConstraintAnd{},
+			expect: "ConstraintAnd",
+		},
+		{
+			input:  gqt.ConstraintMap{},
+			expect: "ConstraintMap",
+		},
+		{
+			input:  gqt.ConstraintAny{},
+			expect: "ConstraintAny",
+		},
+		{
+			input:  gqt.ConstraintTypeEqual{},
+			expect: "ConstraintTypeEqual",
+		},
+		{
+			input:  gqt.ConstraintTypeNotEqual{},
+			expect: "ConstraintTypeNotEqual",
+		},
+		{
+			input:  gqt.ConstraintValEqual{},
+			expect: "ConstraintValEqual",
+		},
+		{
+			input:  gqt.ConstraintValNotEqual{},
+			expect: "ConstraintValNotEqual",
+		},
+		{
+			input:  gqt.ConstraintValGreater{},
+			expect: "ConstraintValGreater",
+		},
+		{
+			input:  gqt.ConstraintValLess{},
+			expect: "ConstraintValLess",
+		},
+		{
+			input:  gqt.ConstraintValGreaterOrEqual{},
+			expect: "ConstraintValGreaterOrEqual",
+		},
+		{
+			input:  gqt.ConstraintValLessOrEqual{},
+			expect: "ConstraintValLessOrEqual",
+		},
+		{
+			input:  gqt.ConstraintBytelenEqual{},
+			expect: "ConstraintBytelenEqual",
+		},
+		{
+			input:  gqt.ConstraintBytelenNotEqual{},
+			expect: "ConstraintBytelenNotEqual",
+		},
+		{
+			input:  gqt.ConstraintBytelenGreater{},
+			expect: "ConstraintBytelenGreater",
+		},
+		{
+			input:  gqt.ConstraintBytelenLess{},
+			expect: "ConstraintBytelenLess",
+		},
+		{
+			input:  gqt.ConstraintBytelenGreaterOrEqual{},
+			expect: "ConstraintBytelenGreaterOrEqual",
+		},
+		{
+			input:  gqt.ConstraintBytelenLessOrEqual{},
+			expect: "ConstraintBytelenLessOrEqual",
+		},
+		{
+			input:  gqt.ConstraintLenEqual{},
+			expect: "ConstraintLenEqual",
+		},
+		{
+			input:  gqt.ConstraintLenNotEqual{},
+			expect: "ConstraintLenNotEqual",
+		},
+		{
+			input:  gqt.ConstraintLenGreater{},
+			expect: "ConstraintLenGreater",
+		},
+		{
+			input:  gqt.ConstraintLenLess{},
+			expect: "ConstraintLenLess",
+		},
+		{
+			input:  gqt.ConstraintLenGreaterOrEqual{},
+			expect: "ConstraintLenGreaterOrEqual",
+		},
+		{
+			input:  gqt.ConstraintLenLessOrEqual{},
+			expect: "ConstraintLenLessOrEqual",
+		},
+	} {
+		t.Run(td.expect, func(t *testing.T) {
+			name := td.input.Name()
+			require.Equal(t, td.expect, name)
+		})
+	}
+}
+
+func TestConstraintKeyAndValue(t *testing.T) {
```

Comment:
This test is missing a case where `input: nil; key: ""; value: nil`. However, remove the test since the function to be tested is to be removed.

Reply:
Didn't understand why we need to test `input: nil`

Reply:
Because we want to test all possible inputs